File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int prefixsum = 0;
        int ans = nums.size() + 1;
        int left = 0;
        
        for (int i = 0; i < nums.size(); i++){
            prefixsum += nums[i];
            while (prefixsum >= target){
                ans = min(ans, i-left+1);
                prefixsum -= nums[left++];
            }
                
        }
        if (ans <= nums.size()) return ans;
        return 0;
    }
};

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        prefixsum = 0
        left, ans = 0, len(nums)+1
        for r, val in enumerate(nums):
            prefixsum += val
            while prefixsum >= target:
                ans = min(ans, r-left+1)
                prefixsum -= nums[left]
                left += 1
        return ans if ans <= len(nums) else 0
        